addOns/graphql/src/test/java/org/zaproxy/addon/graphql/GraphQlFingerprinterUnitTest.java

```diff
@@ -309,21 +309,28 @@ private static String errorResponse(String error, String field, boolean data) {
                 + " }";
     }
 
+    @SuppressWarnings("null")
     @ParameterizedTest
     @MethodSource("fingerprintData")
     void shouldFingerprintValidData(String graphqlImpl, String response) throws Exception {
         // Given
-        ExtensionAlert extensionAlert = mockExtensionAlert();
         nano.addHandler(new GraphQlResponseHandler(response));
         var fp = new GraphQlFingerprinter(UrlBuilder.build(endpointUrl));
+        DiscoveredGraphQlEngine[] discoveredEngine = new DiscoveredGraphQlEngine[1];
+        GraphQlFingerprinter.addEngineHandler(
+                (engine) -> {
+                    discoveredEngine[0] = engine;
+                });
         // When
         fp.fingerprint();
         // Then
-        ArgumentCaptor<Alert> alertArgCaptor = ArgumentCaptor.forClass(Alert.class);
-        verify(extensionAlert).alertFound(alertArgCaptor.capture(), any());
-        Alert alert = alertArgCaptor.getValue();
+        Alert.Builder builder = GraphQlFingerprinter.createFingerprintingAlert(discoveredEngine[0]);
+        Alert alert = builder.build();
         assertThat(alert, is(notNullValue()));
         assertThat(alert.getDescription(), containsString(graphqlImpl));
+        // Check "handled" values
+        assertThat((discoveredEngine[0]).getUri().toString(), is(equalTo(endpointUrl)));
+        assertThat(graphqlImpl, is(equalTo((discoveredEngine[0]).getName())));
```

Comment:
The actual and expected params are interchanged,
```suggestion
        assertThat(discoveredEngine[0].getName(), is(equalTo(graphqlImpl)));
```